mypy/checkmember.py

```diff
@@ -624,12 +624,10 @@ def analyze_class_attribute_access(itype: Instance,
             return mx.chk.handle_partial_var_type(t, mx.is_lvalue, symnode, mx.context)
 
         # Find the class where method/variable was defined.
-        # mypyc hack to workaround mypy misunderstanding multiple inheritance (#3603)
-        node_node = node.node  # type: Any
-        if isinstance(node_node, Decorator):
-            super_info = node_node.var.info  # type: Optional[TypeInfo]
-        elif isinstance(node_node, (Var, FuncBase)):
-            super_info = node_node.info
+        if isinstance(node.node, Decorator):
+            super_info = node.node.var.info  # type: Optional[TypeInfo]
+        elif isinstance(node.node, (Var, SYMBOL_FUNCBASE_TYPES)):
```

Comment:
So the hack now is that this is a variable and the binder doesn't know whether it's true or false?

Reply:
There's no hack, anymore, really. We're just explicitly enumerating the overlap between SymbolNode and FuncBase instead of using isinstance on FuncBase